### PR TITLE
Refactor

### LINKED ISSUE #11: Automatically Generate Event Pages
The event pages (2020, 2019, etc.) are repetitive in format and should be pretty simple to refactor into a dynamic page generation system. The json file containing the necessary information for all the years and their events should be all that needs to be changed to update all the events. As they stand, the event pages are all manually configured.

### PR DESCRIPTION
**Pull request by SamLee514**
Redo the events page/subpages with Gatsby data layer, other minor housekeeping, descriptive variables

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (Chinchuluun1029) [inline .vscode/settings.json]: I think .vscode should be gitignored

Comment (aadibajpai) [reply .vscode/settings.json]: I think it's useful for ppl who use vscode

**Review by aadibajpai — APPROVED** [review]
you're the boss

File: .vscode/settings.json (hunk 1 above)
Comment (SamLee514) [reply .vscode/settings.json]: u very right

Comment (SamLee514) [reply .vscode/settings.json]: @aadibajpai good for consistency in some cases I think but my settings have some personal prefs that might not need to get pushed around

Comment (aadibajpai) [reply .vscode/settings.json]: ah ok I think for vaken they maintain it

File: src/components/events/EventBox.tsx [hunk 2]
```diff
@@ -20,19 +20,43 @@ const T1BG = styled('div')`
...
+     )
+     const info = data.allEventsJson.nodes;
+     const allYears = info.map((year) => {
 
-    const allYears = totalevents.map((year) => {
-      
       const eventsThisYear = year.Information.map((event) => {
         return(
-        <a className="text-container" href={"/years/".concat(year.Year).concat('/#').concat(event.Title)}>
+        <a className="text-container" key={event.Title} href={"/events/".concat(year.Year).concat('/#').concat(event.Title)}>
```
Comment (Chinchuluun1029) [inline src/components/events/EventBox.tsx]: I think using string template literals `/events/${year.Year}` etc would be cleaner than chaining concats here

**Review by Chinchuluun1029 — CHANGES_REQUESTED** [review]
Left a few comments regarding the added code in some files. Didn't fully look at all deleted files, hopefully it runs w/o them.

File: src/components/events/EventBox.tsx (hunk 2 above)
Comment (saydus) [reply src/components/events/EventBox.tsx]: i agree

Comment (SamLee514) [reply src/components/events/EventBox.tsx]: Hm I'm not sure why I didn't do this before. I think for some reason the template didn't work but concats did. But can't reproduce so maybe I was just ded lol

**Review by Chinchuluun1029 — APPROVED** [review]
lgtm

**Pull request merged**